torchvision/transforms/functional_tensor.py

```diff
@@ -399,5 +406,30 @@ def pad(img: Tensor, padding: List[int], fill: int, padding_mode: str = "constan
 
     p = [pad_left, pad_right, pad_top, pad_bottom]
 
+    if padding_mode == "edge":
+        # remap padding_mode str
+        padding_mode = "replicate"
+
+    need_squeeze = False
+    if img.ndim < 4:
+        img = img.unsqueeze(dim=0)
+        need_squeeze = True
+
+    out_dtype = img.dtype
+    need_cast = False
+    if img.dtype not in (torch.float32, torch.float64):
```

Comment:
This could probably be optimized because `constant` padding supports `uint8` types as well.

Reply:
I agree, will fix !